ksqldb-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlTargetUtil.java

```diff
@@ -69,6 +71,63 @@ public static List<StreamedRow> toRows(final Buffer buff) {
     return rows;
   }
 
+  /**
+   * @param buff the buffer response that we get from the /query-stream endpoint
+   *             for KsqlMediaType.KSQL_V1_PROTOBUF
+   * @return a List of StreamedRow that includes the header row as its fist element,
+   *     and every subsequent element is a result row in the protobuf format
+   */
+  public static List<StreamedRow> toRowsFromProto(final Buffer buff) {
+    final List<StreamedRow> rows = new ArrayList<>();
+    boolean headerRead = false;
+
+    // split the buffer across the commas delimiting the individual rows
+    // Since, there would be commas in the header we are replacing
+    // },{"row": with },\n\t\n{"row": and then splitting on \n\t\n to avoid
+    // splitting on commas that might be more that 1 level deep
+    final List<Buffer> buffRows = Arrays
+            .stream(buff
+                    .toString()
+                    .replace("},{\"row\":", "},\n\t\n{\"row\":")
+                    .split("\n\t\n"))
```

Comment:
this seems pretty brittle - I guess we "know" that the contents of `row` are protobuf encoded base64 bytes, right? Is a comma even possible to show up in base64? if it isn't base64, then why can't `\n\t\n` show up as well?

can we use a "real" json parser here to extract each row?

Reply:
using a real json parser now